### ISSUE TITLE
Cart discounts is not applied to updated items

### ISSUE THREAD

**Issue by Reached** [happypixels/laravel-shopr#14]
**Describe the bug**
Im trying to update items in the cart by making a call to the API like this

```javascript
window.axios.patch('/api/shopr/cart/items/' + item.id, {quantity: item.quantity}).then(response => {
      this.cart = response.data;
      console.log(this.cart);
});
```
This works fine, and it returns the cart summary (UI updates etc.), however it does not seem to take into account any discounts that have been applied already. I added a discount with a value of 50.00 as a percentage discount. 

Also I can confirm that the cart summary that comes back from the backend does not reflect the changes either. 🤔 

Here is my full cart summary object for reference (don't mind the formatting):

```json
{  
   "items":[  
      {  
         "id":"15497087885c5eadf485ce3",
         "quantity":4,
         "shoppableType":"App\\Product",
         "shoppableId":1,
         "shoppable":{  
            "id":1,
            "name":"Nice AirMax",
            "description":"This is the most amazing shoe you will ever see lol!",
            "featured_image":"https:\/\/c.static-nike.com\/a\/images\/t_PDP_1280_v1\/f_auto\/gorfwjchoasrrzr1fggt\/air-max-270-shoe-nnTrqDGR.jpg",
            "price":800,
            "created_at":null,
            "updated_at":null
         },
         "options":[  

         ],
         "subItems":[  

         ],
         "total":3200,
         "price":800,
         "price_formatted":"DKK\u00a0800.00"
      }
   ],
   "discounts":[  
      {  
         "id":"15497087935c5eadf961800",
         "quantity":1,
         "shoppableType":"Happypixels\\Shopr\\Models\\DiscountCoupon",
         "shoppableId":2,
         "shoppable":{  
            "id":2,
            "valid_from":"2019-02-08 15:42:34",
            "valid_until":"2019-05-08 15:42:34",
            "uses":8,
            "code":"NEWCODE",
            "description":null,
            "is_fixed":0,
            "value":"50.00",
            "lower_cart_limit":"0.00",
            "created_at":"2019-02-08 15:42:34",
            "updated_at":"2019-02-09 10:37:57",
            "deleted_at":null
         },
         "options":[  

         ],
         "subItems":[  

         ],
         "total":-400,
         "price":-400,
         "price_formatted":"-DKK\u00a0400.00"
      }
   ],
   "sub_total":2240,
   "sub_total_formatted":"DKK\u00a02,240.00",
   "tax_total":560,
   "tax_total_formatted":"DKK\u00a0560.00",
   "total":2800,
   "total_formatted":"DKK\u00a02,800.00",
   "count":4
}
```

Is this intended design perhaps?

**Expected behavior**
The cart total and subtotal should be updated and take discounts into account also.

![cart behavior](https://user-images.githubusercontent.com/5656190/52519809-4ace6180-2c61-11e9-9c7d-c80e266258be.gif)

**Comment by mattias-sanfridsson** [comment]
Thanks for the report! I can confirm this issue, working on a fix.

**Comment by mattias-sanfridsson** [comment]
Should be fixed in 1.2.1. Let me know if you encounter any further issues with this @Reached!

**Comment by Reached** [comment]
Works perfectly, thank you for the quick fix! 👌
